lib/router/index.js

```diff
@@ -281,6 +284,16 @@ proto.handle = function handle(req, res, out) {
     });
   }
 
+  function generateMatchedRoute (prevMatch, curMatch) {
```

Comment:
This is a little verbose, and makes me feel that perhaps an Array structure might be better suited since we can then avoid this logic.